### ISSUE TITLE
GDScript(?) Packed[BASE]Array(?) weird "data sharing"

### ISSUE THREAD

**Issue by Faless** [godotengine/godot#36892]
**Godot version:** Godot 4.0.custom 8cb6d5daa4e41443d658265379b6079d8a087b41 (likely post poolvector removal)

**OS/device including version:**
Ubuntu Linux 18.04.4 LTS

**Issue description:**
In GDScript, packed arrays of base types (e.g. PackedByteArray, PackedFloat32Array, etc), seems to all share the same underlying resource (addressing issue? copy-on-write? ...)

**Steps to reproduce/Minimal reproduction project:**

```gdscript
extends Node2D

func _ready():
	var b1 = PackedByteArray()
	b1.append(2)
	var my_var = PackedByteArray()
	my_var.resize(3)

	var f1 = PackedFloat32Array()
	f1.append(2)
	var f2 = PackedFloat32Array()
	f2.resize(4)

	printt(b1.size(), my_var.size()) # Prints (3, 3) , should be (1, 3)
	printt(f1.size(), f2.size()) # Prints (4, 4) , should be (1, 4)
```

Potentially related to #36611 , #36779

**Comment by nathanfranke** [comment]
Still reproducible a09e03569

**Comment by Faless** [comment]
Fixed via #39100
